src/benchmarks/micro/corefx/System.Runtime/Perf.Enum.cs

```diff
@@ -20,19 +22,534 @@ public enum Colors
+
+        [Benchmark]
+        public ByteEnum ParseNameGenericByte() => Enum.Parse<ByteEnum>("Two");
+
+        [Benchmark]
+        public SByteEnum ParseNameGenericSByte() => Enum.Parse<SByteEnum>("Two");
+
+        [Benchmark]
+        public UInt16Enum ParseNameGenericUInt16() => Enum.Parse<UInt16Enum>("Two");
+
+        [Benchmark]
+        public Int16Enum ParseNameGenericInt16() => Enum.Parse<Int16Enum>("Two");
+
+        [Benchmark]
+        public UInt32Enum ParseNameGenericUInt32() => Enum.Parse<UInt32Enum>("Two");
+
+        [Benchmark]
+        public Int32Enum ParseNameGenericInt32() => Enum.Parse<Int32Enum>("Two");
+
+        [Benchmark]
+        public UInt64Enum ParseNameGenericUInt64() => Enum.Parse<UInt64Enum>("Two");
+
+        [Benchmark]
+        public Int64Enum ParseNameGenericInt64() => Enum.Parse<Int64Enum>("Two");
+
+        [Benchmark]
+        public BigContiguousEnum ParseNameGenericBigContiguous() => Enum.Parse<BigContiguousEnum>("C");
+
+        [Benchmark]
+        public SmallContiguousEnum ParseNameGenericSmallContiguous() => Enum.Parse<SmallContiguousEnum>("One");
+
+        [Benchmark]
+        public ByteEnum ParseValueGenericByte() => Enum.Parse<ByteEnum>("2");
+
+        [Benchmark]
+        public SByteEnum ParseValueGenericSByte() => Enum.Parse<SByteEnum>("2");
+
+        [Benchmark]
+        public UInt16Enum ParseValueGenericUInt16() => Enum.Parse<UInt16Enum>("2");
+
+        [Benchmark]
+        public Int16Enum ParseValueGenericInt16() => Enum.Parse<Int16Enum>("2");
+
+        [Benchmark]
+        public UInt32Enum ParseValueGenericUInt32() => Enum.Parse<UInt32Enum>("2");
+
+        [Benchmark]
+        public Int32Enum ParseValueGenericInt32() => Enum.Parse<Int32Enum>("2");
+
+        [Benchmark]
+        public UInt64Enum ParseValueGenericUInt64() => Enum.Parse<UInt64Enum>("2");
+
+        [Benchmark]
+        public Int64Enum ParseValueGenericInt64() => Enum.Parse<Int64Enum>("2");
+
+        [Benchmark]
+        public bool TryParseOverflowGeneric() => Enum.TryParse<DayOfWeek>("9223372036854775807", out _); // long.MaxValue
+
+        [Benchmark]
+        public bool TryParseMissingGenericBigContiguous() => Enum.TryParse<BigContiguousEnum>("ABC", out _);
+
+        [Benchmark]
+        public bool TryParseMissingGenericSmallContiguous() => Enum.TryParse<SmallContiguousEnum>("ABC", out _);
+
+        [Benchmark]
+        [Arguments(typeof(DayOfWeek))]
+        public bool TryParseOverflowNonGeneric(Type enumType) => Enum.TryParse(enumType, "9223372036854775807", out _); // long.MaxValue
+
+        [Benchmark]
+        [Arguments(typeof(BigContiguousEnum), "ABC")]
+        [Arguments(typeof(SmallContiguousEnum), "ABC")]
+        public bool TryParseMissingNonGeneric(Type enumType, string value) => Enum.TryParse(enumType, value, out _);
+        #endregion
+
+        #region HybridSearchCutoff
+        // Last found cutoff was 32
+
+        //[Benchmark]
```

Comment:
is it intentional that the benchmark is commented out here?

Reply:
Yes, it was used to determine at what length of array a linear search is more performant over a binary search. It's not something that would provide much value except for determining the constant which should probably be manually invoked.